### ISSUE TITLE
Guava 18 is vulnerable. Upgrade to version 24 or newer is recommended

### ISSUE THREAD

**Issue by piskachev** [JesusFreke/smali#686]
Smali depends on guava version 18, which is vulnerable. Please see the vulnerability in this link:
https://snyk.io/vuln/SNYK-JAVA-COMGOOGLEGUAVA-32236

Even though the critical classes (AtomicDoubleArray and CompoundOrdering) are not used in smali, it is recommended to upgrade guava to at least version 24.

**Comment by JesusFreke** [comment]
The main constraint here is that I don't want to upgrade to a version higher than the current AOSP source is using. So I need to take a look and see what version is currently being used there.

**Comment by Lanchon** [comment]
whether smali uses the vulnerable (gadget) classes or not is irrelevant. the mere presence of them in the classpath renders java deserialization that might exist elsewhere vulnerable to shell code execution.

> Mad Gadget is one of the most pernicious vulnerabilities we’ve seen. By merely existing on the Java classpath, seven “gadget” classes in Apache Commons Collections (versions 3.0, 3.1, 3.2, 3.2.1, and 4.0) make object deserialization for the entire JVM process Turing complete with an exec function.

(actually not the entire JVM, but the code running under the affected classloader.)

**Comment by JesusFreke** [comment]
Right, but those classes won't even be included in the binaries, due to proguard.

The main thing is to protect users who link with dexlib2 (and transitively, guava), who don't use proguard.

**Comment by Lanchon** [comment]
yep, i know. that happens to be... me!
